packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -415,24 +415,25 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
         },
       );
 
-      // Register the ability to quickly mark elements as dirty.
-      // The performance of this method may be improved with additional
-      // information from https://github.com/flutter/flutter/issues/46195.
       registerServiceExtension(
         name: 'fastReassemble',
         callback: (Map<String, Object> params) async {
-          final String className = params['class'] as String;
           void markElementsDirty(Element element) {
             if (element == null) {
               return;
             }
-            if (element.widget?.runtimeType?.toString()?.startsWith(className) ?? false) {
+            if (_debugFastReassembleMethod(element.widget)) {
               element.markNeedsBuild();
             }
             element.visitChildElements(markElementsDirty);
           }
+          if (_debugFastReassembleMethod == null) {
+            throw FlutterError('_debugFastReassembleMethod was unexpectedly null.');
+          }
           markElementsDirty(renderViewElement);
-          return <String, String>{'Success': 'true'};
+          await endOfFrame;
```

Comment:
Is there potential for a race condition here? E.g. while we wait asynchronously for the end of the frame, somebody sets debugFastReassembleMethod to a new function, which then incorrectly gets nulled out in the next line?

Reply:
Good idea, I should grab this callback in a local at the start

Reply:
Done